### ISSUE TITLE
Use long polling to poll the CAS for anchors

### ISSUE THREAD

**Issue by stbrody** [ceramicnetwork/js-ceramic#1515]
Currently it takes up to a minute after an anchor is performed before the ceramic node learns about it. Using long polling could make it almost instantaneous. This would improve user experience as well as speed up many of our tests

**Comment by oed** [comment]
Why is this approach preferable over having the anchor service just publish the anchor commit to the stream?

**Comment by stbrody** [comment]
> Why is this approach preferable over having the anchor service just publish the anchor commit to the stream?

If the anchor service publishes the commit, then you have to have the anchor service pin that stream/commit and have some policy for how long it retains it.  You have a similar problem already with the data in the anchor database, but it's a little easier to manage there.  Also, if the CAS publishes the commit, then you need to handle what happens if the Ceramic node that did the update misses the UPDATE message from the CAS ceramic node (since I'm pretty sure gossipsub isn't guaranteed delivery). Maybe you wind up with some sort of polling of QUERY messages, which doesn't feel great.

I think in the end the two approaches are probably roughly equivalent, they just shift where the complexity lives.

**Comment by stbrody** [comment]
Also I guess even making the CAS publish the anchor commit doesn't let you remove the code from Ceramic that polls the CAS for the anchor status, because otherwise you have no way know when your anchor request fails.

**Comment by oed** [comment]
> I think in the end the two approaches are probably roughly equivalent, they just shift where the complexity lives. I don't feel strongly one way or the other as to which is the better path forward

I feel like CAS publishing would be more aligned with the future validator network.

> Also I guess even making the CAS publish the anchor commit doesn't let you remove the code from Ceramic that polls the CAS for the anchor status, because otherwise you have no way know when your anchor request fails.

Interesting point. I wonder how we can deal with this sort of thing in the validator network. I suppose the latest tip would just be in the mempool until it's either rejected or included in a block.

**Comment by stbrody** [comment]
> Interesting point. I wonder how we can deal with this sort of thing in the validator network. I suppose the latest tip would just be in the mempool until it's either rejected or included in a block.

Yeah I imagine even with the validator network the Ceramic node has to do some polling of the validator(s) to get the result of its anchor request. It's analogous to how eth clients have to poll to find out whether their transactions get mined or not.

**Comment by oed** [comment]
I hope we could avoid that kind of separation between the validator network and main ceramic network!
